### PR TITLE
React Hash Routing

### PR DESCRIPTION
**Pull request by htjahyon**
image

### CODE REVIEW

File: react-hash-routing/client/app.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  componentDidMount() {
+    /**
+     * Listen for hash change events on the window object
+     * Each time the window.location.hash changes, parse
+     * it with the parseRoute() function and update state
+     */
+    const route = parseRoute(window.location.hash);
+    this.setState({ route: route });
```
Comment (scbowler) [inline react-hash-routing/client/app.jsx]: You need an event listener here. You should add an event listener to the window listening for a `hashchange` event

**Review by scbowler — CHANGES_REQUESTED** [review]
Everything looks good except for your `componentDidMount` in `app.jsx`, see below comments for details. Let me know if you have any questions

**Comment by htjahyon** [comment]
image

**Review by scbowler — APPROVED** [review]
Looks good! 👍

**Pull request merged**